pkg/codegen/codegen.go

```diff
@@ -363,24 +363,28 @@ func GenerateConstants(t *template.Template, ops []OperationDefinition) (string,
 	return GenerateTemplates([]string{"constants.tmpl"}, t, constants)
 }
 
+// Not the best, but let's do it this way for now.
+// Define a package variable that holds the Schemas to allow us to
+// dereference schemas from another part of the tree.
+var Schemas = make(map[string]*openapi3.SchemaRef)
```

Comment:
This is obviously a considerable change from the existing philosophy, but something is required to allow us to find any other Schema to dereference it. One thing I could do is wrap this in accesor functions rather than just make a package-wide variable. Thoughts?